programs/eosio-blocklog/main.cpp

```diff
@@ -280,6 +320,17 @@ int main(int argc, char** argv) {
          rt.report();
          return 0;
       }
+      if (blog.prune_transactions) {
+         const auto  blocks_dir        = vmap.at("blocks-dir").as<bfs::path>();
+         const auto  state_history_dir = vmap.at("state-history-dir").as<bfs::path>();
+         const auto  block_num         = vmap.at("block-num").as<uint32_t>();
+         const auto  ids               = vmap.at("transaction").as<std::vector<string>>();
```

Comment:
Doesn't this fail badly if none of these are missing. I think you need to add `contains` for each with an appropriate error message.

Reply:
done